src/components/OpenToggle/index.js

```diff
@@ -0,0 +1,24 @@
+import {Component} from 'react'
+
+export default class extends Component {
+  
+  state = {
+    isOpen: false,
+  }
+
+  handleOpenToggleClick = () => {
+    const {isOpen} = this.state
+    this.setState({
+      isOpen: !isOpen,
+    })
+  }
+
+  render() {
+    const {isOpen} = this.state
+    const {children} = this.props
+    return children({
+      isOpen,
+      handleOpenToggleClick: this.handleOpenToggleClick,
+    })
+  }
+}
```

Comment:
I made this new Render Callback component (aka function as child), now that this logic is being shared across more than one component.